fetcher/fedora.go

```diff
@@ -0,0 +1,355 @@
+				cveIDs := []string{}
+				for _, ref := range update.References {
+					var ids []string
+					var err error
+					if isFedoraUpdateInfoTitleReliable(ref.Title) {
+						ids = util.CveIDPattern.FindAllString(ref.Title, -1)
+						if ids == nil {
+							// tyr to correct CVE-ID from description, if title has no CVE-ID
+							// NOTE: If this implementation causes the result of collecting a lot of incorrect information, fix to remove it
+							ids = util.CveIDPattern.FindAllString(update.Description, -1)
+						}
+					} else {
+						ids, err = fetchCveIDsFromBugzilla(ref.ID)
+						if err != nil {
+							return nil, xerrors.Errorf("Failed to fetch CVE-IDs from bugzilla, err: %w", err)
+						}
+					}
+					if ids != nil {
+						cveIDs = append(cveIDs, ids...)
+					}
+				}
+				update.CVEIDs = util.UniqueStrings(cveIDs)
+				securityUpdate = append(securityUpdate, update)
+			}
+		}
+		updateInfo.UpdateList = securityUpdate
+		updateInfos[r.Target] = &updateInfo
+	}
+	return updateInfos, nil
+}
+
+func isFedoraUpdateInfoTitleReliable(title string) bool {
+	if variousFlawsPattern.MatchString(title) {
+		return false
+	}
+	// detect unreliable CVE-ID like CVE-01-0001, CVE-aaa-bbb
+	return len(util.CveIDPattern.FindAllString(title, -1)) == strings.Count(title, "CVE-")
+}
+
+func fetchModuleFeedFilesFedora(reqs []fetchRequest) ([]FetchResult, error) {
+	if len(reqs) == 0 {
+		return nil, xerrors.New("There are no versions to fetch")
+	}
+	results, err := fetchFeedFiles(reqs)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch. err: %w", err)
+	}
+	return results, nil
+}
+
+func fetchModulesYamlFedora(results []FetchResult) (fedoraModuleInfosPerVersion, error) {
+	log15.Info("start fetch modules.yaml in repomd.xml")
+	updateInfoReqs, err := extractInfoFromRepoMd(results, "modules", mimeTypeGzip)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to extract modules from xml, err: %w", err)
+	}
+
+	if len(updateInfoReqs) == 0 {
+		return nil, fmt.Errorf("No updateinfo field in the repomd")
+	}
+
+	results, err = fetchFeedFiles(updateInfoReqs)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch modules.yaml, err: %w", err)
+	}
+
+	yamls := make(fedoraModuleInfosPerVersion, len(results))
+	for _, v := range results {
+		m, err := parseModulesYamlFedora(v.Body)
+		if err != nil {
+			return nil, xerrors.Errorf("Failed to parse modules.yaml, err: %w", err)
+		}
+		yamls[v.Target] = m
+	}
+	return yamls, nil
+}
+
+func parseModulesYamlFedora(b []byte) (fedoraModuleInfosPerPackage, error) {
+	modules := fedoraModuleInfosPerPackage{}
+	scanner := bufio.NewScanner(bytes.NewReader(b))
+	var contents []string
+	for scanner.Scan() {
+		str := scanner.Text()
+		switch str {
+		case "---":
+			{
+				contents = []string{}
+			}
+		case "...":
+			{
+				var module FedoraModuleInfo
+				err := yaml.NewDecoder(strings.NewReader(strings.Join(contents, "\n"))).Decode(&module)
+				if err != nil {
```

Comment:
How about this coding style?
```suggestion
				if err := yaml.NewDecoder(strings.NewReader(strings.Join(contents, "\n"))).Decode(&module); err != nil {
```